internal/runners/projects/projects.go

```diff
@@ -55,19 +64,48 @@ func (r *Projects) fetchProjects() ([]projectWithOrg, *failures.Failure) {
 		}
 		return nil, api.FailUnknown.Wrap(err)
 	}
-	projectsList := []projectWithOrg{}
+	projects := []projectWithOrg{}
+	localConfigProjects := r.config.GetStringMapStringSlice(projectfile.LocalProjectsConfigKey)
 	for _, org := range orgs.Payload {
-		orgProjects, err := model.FetchOrganizationProjects(org.URLname)
+		platformOrgProjects, err := model.FetchOrganizationProjects(org.URLname)
 		if err != nil {
 			return nil, err
 		}
-		for _, project := range orgProjects {
-			desc := ""
-			if project.Description != nil {
-				desc = *project.Description
+
+		orgProjects := make([]projectWithOrg, len(platformOrgProjects))
+		for i, project := range platformOrgProjects {
+			p := projectWithOrg{
+				Name:         project.Name,
+				Organization: org.Name,
+			}
+
+			// Description can be non-nil but also empty
+			if project.Description != nil && *project.Description != "" {
+				p.Name = nameAndDescription(project.Name, *project.Description)
+			}
+
+			// Viper lowers all map keys so we must do the same here
+			// in order to retrieve the locally cached projects
+			if localPaths, ok := localConfigProjects[fmt.Sprintf("%s/%s", strings.ToLower(org.URLname), strings.ToLower(project.Name))]; ok {
+				p.LocalCheckouts = localPaths
 			}
-			projectsList = append(projectsList, projectWithOrg{project.Name, desc, org.Name})
+			orgProjects[i] = p
 		}
+
+		sort.Slice(orgProjects, func(i, j int) bool {
+			return orgProjects[i].LocalCheckouts != nil && orgProjects[j].LocalCheckouts == nil
+		})
```

Comment:
Run this outside of the loop on `projects`. No point sorting this every time an entry is added, only need to sort it once.

Reply:
Running this particular sort outside of the organizations loop will move all of the projects to the top of the table. One of the ACs was to keep the overall ordering of the table based on organizations.

Reply:
Let's make it:

```
return (orgProjects[i].LocalCheckouts != nil && orgProjects[j].LocalCheckouts == nil) ||
	orgProjects[i].Organization < orgProjects[j].Organization ||
	orgProjects[i].Name < orgProjects[i].Name
```

And move it outside the loop. That way it stays performant, the AC's are handled AND we get name sorting as a bonus.

Reply:
I modified your suggestion slightly to maintain the organization ordering. It is not sorted by name within the organizations but should maintain how we received it from the platform.

Reply:
After testing this further it also can also break the organization ordering. I think we need to decide which ordering we want. 

If it should be ordered by organization then the initial sorting can be kept. The sort will only be called once for every organization that a user is a part of, which for most users I imagine won't be very high.

If we want the local projects at the top of the list and all others below that we can do one sort which should handle it.

Reply:
Let's not get hung up on it for this PR seeing as it's proving to be a challenge. Checkouts should be sorted on top as per the AC, the rest will have to follow in a separate story.